pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/AuxClassPathController.java

```diff
@@ -0,0 +1,183 @@
+    private Button removeFileButton;
+    @FXML
+    private Button selectFilesButton;
+    @FXML
+    private ListView<File> fileListView;
+    @FXML
+    private Button moveItemUpButton;
+    @FXML
+    private Button moveItemDownButton;
+    @FXML
+    private Button setClassPathButton;
+
+
+    public AuxClassPathController(DesignerRoot designerRoot) {
+        this.designerRoot = designerRoot;
+    }
+
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+
+        selectFilesButton.setOnAction(e -> onSelectFileClicked());
+        removeFileButton.setOnAction(e -> onRemoveFileClicked());
+        setClassPathButton.setOnAction(e -> {
+            try {
+                setClassPath(classPathGenerator());
+            } catch (IOException e1) {
+                e1.printStackTrace();
+            }
+        });
+        moveItemUpButton.setOnAction(e -> moveUp());
+        moveItemDownButton.setOnAction(e -> moveDown());
+    }
+
+
+    private void onSelectFileClicked() {
+        FileChooser chooser = new FileChooser();
+        chooser.setTitle("Select Files");
+        chooser.getExtensionFilters().addAll(
+            new FileChooser.ExtensionFilter("Java JARs", "*.jar"),
+            new FileChooser.ExtensionFilter("Java WARs", "*.war"),
+            new FileChooser.ExtensionFilter("Java EARs", "*.ear"),
+            new FileChooser.ExtensionFilter("Java class files", "*.class")
+        );
+        List<File> file = chooser.showOpenMultipleDialog(designerRoot.getMainStage());
+        for (File f : file) {
+            fileListView.getItems().add(f);
+        }
+    }
+
+
+    private void onRemoveFileClicked() {
+        File f = fileListView.getSelectionModel().getSelectedItem();
+        fileListView.getItems().remove(f);
+    }
+
+
+    private void moveUp() {
+        moveItem(-1);
+    }
+
+
+    private void moveDown() {
+        moveItem(1);
+    }
+
+
+    public void moveItem(int direction) {
+        // Checking selected item
+        if (fileListView.getSelectionModel().getSelectedItem() == null) {
+            return;
+        }
+
+        // Calculate new index using move direction
+        int newIndex = fileListView.getSelectionModel().getSelectedIndex() + direction;
+
+        if (newIndex < 0 || newIndex >= fileListView.getItems().size()) {
+            return;
+        }
+
+        File selected = fileListView.getSelectionModel().getSelectedItem();
+
+        // Removing removable element
+        fileListView.getItems().remove(selected);
+        // Insert it in new position
+        fileListView.getItems().add(newIndex, selected);
+        //Restore Selection
+        fileListView.scrollTo(newIndex);
+        fileListView.getSelectionModel().select(newIndex);
+
+    }
+
+
+    private String classPathGenerator() throws IOException {
+
+        String classPath = "";
+
+        for (File f : fileListView.getItems()) {
+            classPath = classPath + ";" + f.getAbsolutePath();
```

Comment:
This is not portable, instead of `";"` you should use `File.pathSeparator`. The separator char on Unix systems is `:`

Reply:
since this module uses Java 8, we can actually use streams here rather than looping over the list and creating new string instance all over:

`fileListView.getItems().stream().collect(Collectors.joining(File.pathSeparator))`